src/lightning_app/cli/cmd_apps.py

```diff
@@ -0,0 +1,81 @@
+import json
+from datetime import datetime
+
+from lightning_cloud.openapi import Externalv1LightningappInstance, V1LightningappInstanceState
+from rich.console import Console
+from rich.table import Table
+from rich.text import Text
+
+from lightning_app.cli.core import Formatable
+from lightning_app.utilities.cloud import _get_project
+from lightning_app.utilities.network import LightningClient
+
+
+class AppManager:
+    """AppManager implements API calls specific to Lightning AI BYOC apps."""
+
+    def __init__(self):
+        self.api_client = LightningClient()
+
+    def list(self, cluster_id=None):
+        project = _get_project(self.api_client)
+
+        args = {
+            "project_id": project.project_id,
+        }
+        if cluster_id is not None:
+            args["cluster_id"] = cluster_id
+
+        resp = self.api_client.lightningapp_instance_service_list_lightningapp_instances(**args)
```

Comment:
we should probably manager exceptions here. Even if not exhaustive, atleast basic ones like cluster permission denied etc - super useful when users make spelling mistakes on cluster id

Reply:
I'll check what typed exceptions we have in our OpenAPI model today and see if I can catch some meaningful errors here :+1:

Reply:
just did some validation on the API - unknown clusters just return an empty response with no error. The API can only return two errors: permission denied when you don't have access to the project, and internal error if anything in the service failed.

because we don't expose project management today, the only way to select a project is via `LIGHTNING_CLOUD_PROJECT_ID` environment variable, in which case `_get_project` already provides a useful error message: 

```
ValueError: Environment variable LIGHTNING_CLOUD_PROJECT_ID is set but could not find an associated project.
```

Let me add a similar error message if the cluster doesn't exist, so at least the CLI is similar in behaviour.

Reply:
hm, implementing this leads to a problem when a cluster is deleted:
we don't delete app releases associated with a cluster, so if I implement a similar validation you reach inconsistent behaviour:

`lightning list apps` will show apps for a cluster that is deleted, but `lightning list apps --cluster-id <deleted-cluster>` will fail because the cluster was deleted.

I'll leave it as it is.